docs/sphinx_extensions.py

```diff
@@ -119,10 +153,20 @@ def run(self) -> List[nodes.Node]:  # noqa: C901
                     else:
                         exec(last_line, exec_globals)
 
-        except Exception as e:
-            error_msg = f"Error executing code: {e}"
-            error_node = nodes.error("", nodes.paragraph(text=error_msg))
-            return [code_node, error_node]
+        except (KeyboardInterrupt, SystemExit):
+            raise
+        except BaseException as e:
+            if expected_error:
+                if e.__class__.__name__ == expected_error:
+                    return [code_node, ErrorMessageNode(f"{e}")]
+                raise
+            raise
+
+        if expected_error:
+            return [
+                code_node,
+                ErrorMessageNode(f"Expected error '{expected_error}' did not occur."),
+            ]
```

Comment:
If the expected error did not occur, this should raise an error instead of the error message being baked into the docs